pkg/apis/controller/.gitignore

```diff
@@ -0,0 +1,2 @@
+*
+!.gitignore
```

Comment:
Why introduce those `.gitignore` files?

Reply:
Per your past recommendation to submit smaller sized PRs to improve merge consume-ability this first PR submitted for your review creates isolated directories to contain the extended queuejob thus the gitignore files are needed.  As we submit future PRs to add to these directories the gitinore files will be adjusted/removed.

Reply:
but .gitignore does not make sense to me.

Reply:
Was you question about the content of the gitignore file?  My understanding to make a directory empty (in the repository) that will checkout after a merge is to create a .gitignore file inside that empty directory that contains those lines.  As we submit future PRs to add to these directories the gitinore files will be adjusted/removed.  If you prefer not to have initial directories empty on the first PR I can add a few of the files that will be within each directory to keep the PR small/med in size and remove the gitignore file.  Please advise.